### PR TITLE
Fix Portallocator Testcases

### LINKED ISSUE #4390: Flaky Unit Test: TestPortRangeAllocatorSyncDeleteGameServer
* https://console.cloud.google.com/cloud-build/builds/147dcfeb-96f4-44b0-a859-5f058ccfc006;step=19?project=agones-images
* https://github.com/googleforgames/agones/pull/4326

```
{"message":"Wait for cache sync","severity":"info","time":"2025-12-18T01:53:01.300117927Z"}
{"message":"unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:\"\", Verb:\"watch\", Resource:schema.GroupVersionResource{Group:\"agones.dev\", Version:\"v1\", Resource:\"gameservers\"}, Subresource:\"\"}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:\"\"}, ListOptions:v1.ListOptions{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, LabelSelector:\"\", FieldSelector:\"\", Watch:true, AllowWatchBookmarks:true, ResourceVersion:\"\", ResourceVersionMatch:\"\", TimeoutSeconds:(*int64)(0xc0005121b8), Limit:0, Continue:\"\", SendInitialEvents:(*bool)(nil)}}","severity":"error","time":"2025-12-18T01:53:01.330178102Z"}
{"message":"unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:\"\", Verb:\"watch\", Resource:schema.GroupVersionResource{Group:\"\", Version:\"v1\", Resource:\"nodes\"}, Subresource:\"\"}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:\"\"}, ListOptions:v1.ListOptions{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, LabelSelector:\"\", FieldSelector:\"\", Watch:true, AllowWatchBookmarks:true, ResourceVersion:\"\", ResourceVersionMatch:\"\", TimeoutSeconds:(*int64)(0xc00029e918), Limit:0, Continue:\"\", SendInitialEvents:(*bool)(nil)}}","severity":"error","time":"2025-12-18T01:53:01.336392319Z"}
{"i":1,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.408275581Z","x":0}
{"i":1,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.409084437Z","x":1}
{"i":1,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.409478598Z","x":2}
{"i":1,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.409839686Z","x":3}
{"i":1,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.410333624Z","x":4}
{"i":1,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.410729942Z","x":5}
{"i":1,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.411247676Z","x":6}
{"i":3,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.411530011Z","x":0}
{"i":3,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.415717388Z","x":1}
{"i":4,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.415288Z","x":0}
{"i":4,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.416492333Z","x":1}
{"i":4,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.416995859Z","x":2}
{"i":4,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.417310957Z","x":3}
{"i":4,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.417744307Z","x":4}
{"i":0,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.411431458Z","x":0}
{"i":0,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.418957801Z","x":1}
{"i":0,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.419443652Z","x":2}
{"i":0,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.419889107Z","x":3}
{"i":0,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.420325529Z","x":4}
{"i":0,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.420804429Z","x":5}
{"i":0,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.421269348Z","x":6}
{"i":0,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.421640995Z","x":7}
{"i":0,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.422078302Z","x":8}
{"i":0,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.422516723Z","x":9}
{"i":4,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.418246417Z","x":5}
{"i":4,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.42303768Z","x":6}
{"i":2,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.411300156Z","x":0}
{"i":2,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.423889263Z","x":1}
{"i":2,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.424282217Z","x":2}
{"i":2,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.424672492Z","x":3}
{"i":2,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.425049353Z","x":4}
{"i":2,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.425367347Z","x":5}
{"i":2,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.425954291Z","x":6}
{"i":2,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.426276063Z","x":7}
{"i":2,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.426734476Z","x":8}
{"i":3,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.41610971Z","x":2}
{"i":4,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.427811709Z","x":7}
{"i":4,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.435927564Z","x":8}
{"i":4,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.436482305Z","x":9}
{"i":3,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.427822515Z","x":3}
{"i":3,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.438217961Z","x":4}
{"i":3,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.43871888Z","x":5}
{"i":3,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.43918493Z","x":6}
{"i":3,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.439637512Z","x":7}
{"i":3,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.440205056Z","x":8}
{"i":3,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.440879389Z","x":9}
{"i":1,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.411667233Z","x":7}
{"i":1,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.441692221Z","x":8}
{"i":1,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.442058076Z","x":9}
{"i":2,"message":"allocating!","severity":"info","time":"2025-12-18T01:53:01.427385738Z","x":9}
{"message":"Wait for cache sync","severity":"info","time":"2025-12-18T01:53:01.446306792Z"}
{"message":"Wait for cache sync","severity":"info","time":"2025-12-18T01:53:01.44810997Z"}
{"message":"Wait for cache sync","severity":"info","time":"2025-12-18T01:53:01.449735612Z"}
{"message":"unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:\"\", Verb:\"watch\", Resource:schema.GroupVersionResource{Group:\"agones.dev\", Version:\"v1\", Resource:\"gameservers\"}, Subresource:\"\"}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:\"\"}, ListOptions:v1.ListOptions{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, LabelSelector:\"\", FieldSelector:\"\", Watch:true, AllowWatchBookmarks:true, ResourceVersion:\"\", ResourceVersionMatch:\"\", TimeoutSeconds:(*int64)(0xc0000128b8), Limit:0, Continue:\"\", SendInitialEvents:(*bool)(nil)}}","severity":"error","time":"2025-12-18T01:53:01.457608936Z"}
{"message":"Wait for cache sync","severity":"info","time":"2025-12-18T01:53:01.449802564Z"}
{"message":"Wait for cache sync","severity":"info","time":"2025-12-18T01:53:01.449992501Z"}
{"message":"unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:\"\", Verb:\"watch\", Resource:schema.GroupVersionResource{Group:\"\", Version:\"v1\", Resource:\"nodes\"}, Subresource:\"\"}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:\"\"}, ListOptions:v1.ListOptions{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, LabelSelector:\"\", FieldSelector:\"\", Watch:true, AllowWatchBookmarks:true, ResourceVersion:\"\", ResourceVersionMatch:\"\", TimeoutSeconds:(*int64)(0xc000274298), Limit:0, Continue:\"\", SendInitialEvents:(*bool)(nil)}}","severity":"error","time":"2025-12-18T01:53:01.481232589Z"}
{"message":"unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:\"\", Verb:\"watch\", Resource:schema.GroupVersionResource{Group:\"agones.dev\", Version:\"v1\", Resource:\"gameservers\"}, Subresource:\"\"}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:\"\"}, ListOptions:v1.ListOptions{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, LabelSelector:\"\", FieldSelector:\"\", Watch:true, AllowWatchBookmarks:true, ResourceVersion:\"\", ResourceVersionMatch:\"\", TimeoutSeconds:(*int64)(0xc0004a4b48), Limit:0, Continue:\"\", SendInitialEvents:(*bool)(nil)}}","severity":"error","time":"2025-12-18T01:53:01.482860882Z"}
{"message":"unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:\"\", Verb:\"watch\", Resource:schema.GroupVersionResource{Group:\"agones.dev\", Version:\"v1\", Resource:\"gameservers\"}, Subresource:\"\"}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:\"\"}, ListOptions:v1.ListOptions{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, LabelSelector:\"\", FieldSelector:\"\", Watch:true, AllowWatchBookmarks:true, ResourceVersion:\"\", ResourceVersionMatch:\"\", TimeoutSeconds:(*int64)(0xc0002743b8), Limit:0, Continue:\"\", SendInitialEvents:(*bool)(nil)}}","severity":"error","time":"2025-12-18T01:53:01.49917692Z"}
{"message":"unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:\"\", Verb:\"watch\", Resource:schema.GroupVersionResource{Group:\"\", Version:\"v1\", Resource:\"nodes\"}, Subresource:\"\"}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:\"\"}, ListOptions:v1.ListOptions{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, LabelSelector:\"\", FieldSelector:\"\", Watch:true, AllowWatchBookmarks:true, ResourceVersion:\"\", ResourceVersionMatch:\"\", TimeoutSeconds:(*int64)(0xc000512588), Limit:0, Continue:\"\", SendInitialEvents:(*bool)(nil)}}","severity":"error","time":"2025-12-18T01:53:01.527299524Z"}
{"message":"unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:\"\", Verb:\"watch\", Resource:schema.GroupVersionResource{Group:\"\", Version:\"v1\", Resource:\"nodes\"}, Subresource:\"\"}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:\"\"}, ListOptions:v1.ListOptions{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, LabelSelector:\"\", FieldSelector:\"\", Watch:true, AllowWatchBookmarks:true, ResourceVersion:\"\", ResourceVersionMatch:\"\", TimeoutSeconds:(*int64)(0xc00029e978), Limit:0, Continue:\"\", SendInitialEvents:(*bool)(nil)}}","severity":"error","time":"2025-12-18T01:53:01.527655611Z"}
{"message":"unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:\"\", Verb:\"watch\", Resource:schema.GroupVersionResource{Group:\"agones.dev\", Version:\"v1\", Resource:\"gameservers\"}, Subresource:\"\"}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:\"\"}, ListOptions:v1.ListOptions{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, LabelSelector:\"\", FieldSelector:\"\", Watch:true, AllowWatchBookmarks:true, ResourceVersion:\"\", ResourceVersionMatch:\"\", TimeoutSeconds:(*int64)(0xc0004a4cd8), Limit:0, Continue:\"\", SendInitialEvents:(*bool)(nil)}}","severity":"error","time":"2025-12-18T01:53:01.512089058Z"}
{"message":"Wait for cache sync","severity":"info","time":"2025-12-18T01:53:01.555240689Z"}
{"message":"unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:\"\", Verb:\"watch\", Resource:schema.GroupVersionResource{Group:\"agones.dev\", Version:\"v1\", Resource:\"gameservers\"}, Subresource:\"\"}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:\"\"}, ListOptions:v1.ListOptions{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, LabelSelector:\"\", FieldSelector:\"\", Watch:true, AllowWatchBookmarks:true, ResourceVersion:\"\", ResourceVersionMatch:\"\", TimeoutSeconds:(*int64)(0xc000248238), Limit:0, Continue:\"\", SendInitialEvents:(*bool)(nil)}}","severity":"error","time":"2025-12-18T01:53:01.557567904Z"}
{"message":"Wait for cache sync","severity":"info","time":"2025-12-18T01:53:01.588072899Z"}
{"message":"unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:\"\", Verb:\"watch\", Resource:schema.GroupVersionResource{Group:\"agones.dev\", Version:\"v1\", Resource:\"gameservers\"}, Subresource:\"\"}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:\"\"}, ListOptions:v1.ListOptions{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, LabelSelector:\"\", FieldSelector:\"\", Watch:true, AllowWatchBookmarks:true, ResourceVersion:\"\", ResourceVersionMatch:\"\", TimeoutSeconds:(*int64)(0xc0002485b8), Limit:0, Continue:\"\", SendInitialEvents:(*bool)(nil)}}","severity":"error","time":"2025-12-18T01:53:01.59032042Z"}
{"message":"Wait for cache sync","severity":"info","time":"2025-12-18T01:53:01.668704602Z"}
{"message":"unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:\"\", Verb:\"watch\", Resource:schema.GroupVersionResource{Group:\"agones.dev\", Version:\"v1\", Resource:\"gameservers\"}, Subresource:\"\"}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:\"\"}, ListOptions:v1.ListOptions{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, LabelSelector:\"\", FieldSelector:\"\", Watch:true, AllowWatchBookmarks:true, ResourceVersion:\"\", ResourceVersionMatch:\"\", TimeoutSeconds:(*int64)(0xc000248ad8), Limit:0, Continue:\"\", SendInitialEvents:(*bool)(nil)}}","severity":"error","time":"2025-12-18T01:53:01.676177123Z"}
{"message":"Allocated Port","ports":[{"range":"default","portPolicy":"Dynamic","containerPort":7777,"hostPort":10},{"name":"another","range":"default","portPolicy":"Dynamic","containerPort":6666,"hostPort":13},{"name":"static","range":"default","portPolicy":"Static","containerPort":6666,"hostPort":9999},{"name":"passthrough","range":"default","portPolicy":"Passthrough","containerPort":15,"hostPort":15}],"severity":"info","time":"2025-12-18T01:53:01.771914864Z","uid":"0:0"}
{"message":"Allocated Port","ports":[{"range":"default","portPolicy":"Dynamic","containerPort":7777,"hostPort":11},{"name":"another","range":"default","portPolicy":"Dynamic","containerPort":6666,"hostPort":12},{"name":"static","range":"default","portPolicy":"Static","containerPort":6666,"hostPort":9999},{"name":"passthrough","range":"default","portPolicy":"Passthrough","containerPort":14,"hostPort":14}],"severity":"info","time":"2025-12-18T01:53:01.77335006Z","uid":"0:1"}
{"message":"Allocated Port","ports":[{"range":"default","portPolicy":"Dynamic","containerPort":7777,"hostPort":16},{"name":"another","range":"default","portPolicy":"Dynamic","containerPort":6666,"hostPort":17},{"name":"static","range":"default","portPolicy":"Static","containerPort":6666,"hostPort":9999},{"name":"passthrough","range":"default","portPolicy":"Passthrough","containerPort":20,"hostPort":20}],"severity":"info","time":"2025-12-18T01:53:01.773758878Z","uid":"0:2"}
{"message":"Allocated Port","ports":[{"range":"default","portPolicy":"Dynamic","containerPort":7777,"hostPort":18},{"name":"another","range":"default","portPolicy":"Dynamic","containerPort":6666,"hostPort":19},{"name":"static","range":"default","portPolicy":"Static","containerPort":6666,"hostPort":9999},{"name":"passthrough","range":"default","portPolicy":"Passthrough","containerPort":20,"hostPort":20}],"severity":"info","time":"2025-12-18T01:53:01.774106469Z","uid":"1:0"}
{"message":"Allocated Port","ports":[{"range":"default","portPolicy":"Dynamic","containerPort":7777,"hostPort":18},{"name":"another","range":"default","portPolicy":"Dynamic","containerPort":6666,"hostPort":10},{"name":"static","range":"default","portPolicy":"Static","containerPort":6666,"hostPort":9999},{"name":"passthrough","range":"default","portPolicy":"Passthrough","containerPort":11,"hostPort":11}],"severity":"info","time":"2025-12-18T01:53:01.77449422Z","uid":"1:1"}
{"message":"Allocated Port","ports":[{"range":"default","portPolicy":"Dynamic","containerPort":7777,"hostPort":12},{"name":"another","range":"default","portPolicy":"Dynamic","containerPort":6666,"hostPort":13},{"name":"static","range":"default","portPolicy":"Static","containerPort":6666,"hostPort":9999},{"name":"passthrough","range":"default","portPolicy":"Passthrough","containerPort":16,"hostPort":16}],"severity":"info","time":"2025-12-18T01:53:01.774957961Z","uid":"1:2"}
{"Allocated":18,"Total":22,"message":"Ports count","severity":"info","time":"2025-12-18T01:53:01.779535478Z"}
{"Allocated":21,"Total":22,"message":"Ports count","severity":"info","time":"2025-12-18T01:53:01.779959789Z"}
{"Allocated":24,"Total":33,"message":"Ports count","severity":"info","time":"2025-12-18T01:53:01.780365807Z"}
{"message":"Wait for cache sync","severity":"info","time":"2025-12-18T01:53:01.782007263Z"}
{"message":"unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:\"\", Verb:\"watch\", Resource:schema.GroupVersionResource{Group:\"agones.dev\", Version:\"v1\", Resource:\"gameservers\"}, Subresource:\"\"}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:\"\"}, ListOptions:v1.ListOptions{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, LabelSelector:\"\", FieldSelector:\"\", Watch:true, AllowWatchBookmarks:true, ResourceVersion:\"\", ResourceVersionMatch:\"\", TimeoutSeconds:(*int64)(0xc00029f178), Limit:0, Continue:\"\", SendInitialEvents:(*bool)(nil)}}","severity":"error","time":"2025-12-18T01:53:01.783935436Z"}
{"message":"unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:\"\", Verb:\"watch\", Resource:schema.GroupVersionResource{Group:\"\", Version:\"v1\", Resource:\"nodes\"}, Subresource:\"\"}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:\"\"}, ListOptions:v1.ListOptions{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, LabelSelector:\"\", FieldSelector:\"\", Watch:true, AllowWatchBookmarks:true, ResourceVersion:\"\", ResourceVersionMatch:\"\", TimeoutSeconds:(*int64)(0xc00029f1f8), Limit:0, Continue:\"\", SendInitialEvents:(*bool)(nil)}}","severity":"error","time":"2025-12-18T01:53:01.784916751Z"}
{"message":"Wait for cache sync","severity":"info","time":"2025-12-18T01:53:01.885161533Z"}
{"message":"unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:\"\", Verb:\"watch\", Resource:schema.GroupVersionResource{Group:\"agones.dev\", Version:\"v1\", Resource:\"gameservers\"}, Subresource:\"\"}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:\"\"}, ListOptions:v1.ListOptions{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, LabelSelector:\"\", FieldSelector:\"\", Watch:true, AllowWatchBookmarks:true, ResourceVersion:\"\", ResourceVersionMatch:\"\", TimeoutSeconds:(*int64)(0xc00029f858), Limit:0, Continue:\"\", SendInitialEvents:(*bool)(nil)}}","severity":"error","time":"2025-12-18T01:53:01.8878336Z"}
--- FAIL: TestPortRangeAllocatorSyncDeleteGameServer (1.13s)
    portallocator_test.go:603: 
        	Error Trace:	/go/src/agones.dev/agones/pkg/portallocator/portallocator_test.go:603
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestPortRangeAllocatorSyncDeleteGameServer
FAIL
FAIL	agones.dev/agones/pkg/portallocator	2.162s
```

### PR DESCRIPTION
**Pull request by Sivasankaran25**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug

/kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #4390

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/metrics/controller_test.go [hunk 1]
```diff
@@ -597,6 +597,7 @@ func TestControllerGameServersNodeState(t *testing.T) {
 
 	// wait until we have some nodes and gameservers in metrics
 	var exporter *metricExporter
+	time.Sleep(5 * time.Second)
```
Comment (markmandel) [inline pkg/metrics/controller_test.go]: Sleep feels weird here - can we wait for some kind of state to show up?

Sleep always worries me, like we're hiding something and it's not determinstic.

File: pkg/portallocator/portallocator_test.go [hunk 2]
```diff
@@ -607,8 +614,12 @@ func TestPortRangeAllocatorSyncDeleteGameServer(t *testing.T) {
 	// delete the currently non allocated server, all should be the same
 	// simulated getting an old delete message
 	gsWatch.Delete(gs4.DeepCopy())
+	time.Sleep(5 * time.Second)
```
Comment (markmandel) [inline pkg/portallocator/portallocator_test.go]: Rather than sleep - can we do a check with an Eventually?

Sleep is non deterministic, so this always a concern for future flakes.

File: pkg/metrics/controller_test.go (hunk 1 above)
Comment (Sivasankaran25) [reply pkg/metrics/controller_test.go]: Updated

File: pkg/portallocator/portallocator_test.go (hunk 2 above)
Comment (Sivasankaran25) [reply pkg/portallocator/portallocator_test.go]: Updated

File: pkg/portallocator/portallocator_test.go [hunk 3]
```diff
@@ -570,17 +570,30 @@ func TestPortRangeAllocatorSyncDeleteGameServer(t *testing.T) {
 		return true, nl, nil
 	})
 
-	_, cancel := agtesting.StartInformers(m, pa.gameServerSynced, pa.nodeSynced)
+	ctx, cancel := agtesting.StartInformers(m, pa.gameServerSynced, pa.nodeSynced)
 	defer cancel()
 
+	m.KubeInformerFactory.Start(ctx.Done())
```
Comment (markmandel) [inline pkg/portallocator/portallocator_test.go]: Both starting the informer factories and wating for cache syncs happens within `agtesting.StartInformers` - so why do it again?

File: pkg/portallocator/portallocator_test.go [hunk 4]
```diff
@@ -570,17 +570,30 @@ func TestPortRangeAllocatorSyncDeleteGameServer(t *testing.T) {
...
+		pa.nodeSynced,
+	))
+
 	gsWatch.Add(gs1.DeepCopy())
 	gsWatch.Add(gs2.DeepCopy())
 	gsWatch.Add(gs3.DeepCopy())
 	require.Eventually(t, func() bool {
 		list, err := pa.gameServerLister.GameServers(gs1.ObjectMeta.Namespace).List(labels.Everything())
+		if err != nil || list == nil {
+			return false
```
Comment (markmandel) [inline pkg/portallocator/portallocator_test.go:591]: Two things:

1) Would be nice to have an explanation in the commit / comment why this solves the problem. What was the race condition?
2) Be good to log which version it was that failed (err? list is nill?) , so if it continues to fail we have extra debugging.

**Comment by markmandel** [comment]
Since you didn't say - curious if you did some kind of manual test to work out if this fixes the flake? Run it a few hundred times?

File: pkg/portallocator/portallocator_test.go (hunk 3 above)
Comment (Sivasankaran25) [reply pkg/portallocator/portallocator_test.go]: Yes, this is already implemented, so I am removing this

**Comment by Sivasankaran25** [comment]
> Since you didn't say - curious if you did some kind of manual test to work out if this fixes the flake? Run it a few hundred times?

Yes, I have tested the test cases—not 100 times, but I checked all corner cases and ran both individual and all test cases, and they work correctly

File: pkg/portallocator/portallocator_test.go (hunk 4 above)
Comment (Sivasankaran25) [reply pkg/portallocator/portallocator_test.go:591]: Cache may not be ready yet—returning false lets Eventually() retry safely

File: pkg/portallocator/portallocator_test.go [hunk 5]
```diff
@@ -570,17 +570,29 @@ func TestPortRangeAllocatorSyncDeleteGameServer(t *testing.T) {
 		return true, nl, nil
 	})
 
-	_, cancel := agtesting.StartInformers(m, pa.gameServerSynced, pa.nodeSynced)
+	ctx, cancel := agtesting.StartInformers(m, pa.gameServerSynced, pa.nodeSynced)
 	defer cancel()
 
+	// wait for caches
+	require.True(t, cache.WaitForCacheSync(
```
Comment (markmandel) [inline pkg/portallocator/portallocator_test.go]: We don't need this `cache.WaitForCacheSync` since that's also part of `StartInformers` above.
